File: dp_labs_and_solutions/dp_labs_and_solutions/dp_matlab_labs_and_solutions/abstract_factory/solution/OrganicGarden.m
% OrganicGarden.m

classdef OrganicGarden < Garden
	methods
		% constructor:
		function theOrganicGarden = OrganicGarden(aLettucePrototype)
			if ~isa(aLettucePrototype, 'OrganicLettuce')
				error('An OrganicGarden must have an OrganicLettuce prototype.');
			end
			theOrganicGarden@Garden(aLettucePrototype);
		end

		function result = makeTomato(theGarden)
			result = OrganicTomato(1, 2);
		end
	end
end

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/singleton/abstract_factory/lab/organic_garden.cpp
// organic_garden.cpp

#include "organic_garden.h"
#include "organic_lettuce.h"
#include "organic_tomato.h"
#include "garden.h"

organic_garden::organic_garden(organic_lettuce * a_lettuce_prototype)
: garden(a_lettuce_prototype)
{
}


organic_tomato * organic_garden::make_tomato() const
{
	return new organic_tomato(1, 2);
}